src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -120,10 +127,14 @@ export class FileService implements IFileService {
 		}
 
 		if (this.basePath && !this.options.disableWatcher) {
-			if (isWindows) {
-				this.setupWin32WorkspaceWatching();
+			if (this.options.useNsfwFileWatcher) {
```

Comment:
@bpasero also `getWorkspace2` seemed to be returning a single root at the time of initialization. Maybe I wasn't using it right but that may complicate things.